mlflow/lightgbm.py

```diff
@@ -225,6 +232,42 @@ def callback(env):
                 eval_results.append(res)
             return callback
 
+        def plot_importance(features, importance, importance_type, max_num_features=None):
+            """
+            Plot feature importance.
+            """
+
+            if max_num_features is None:
+                indices = np.argsort(importance)
+            else:
+                indices = np.argsort(importance)[-max_num_features:]
+
+            features = np.array(features)[indices]
+            importance = importance[indices]
+            num_features = len(features)
+
+            # If num_features > 10, increase the figure height to prevent the plot
+            # from being too dense.
+            w, h = [6.4, 4.8]  # matplotlib's default figure size
+            h = h + 0.1 * num_features if num_features > 10 else h
+            fig, ax = plt.subplots(figsize=(w, h))
+
+            yloc = np.arange(num_features)
+            ax.barh(yloc, importance, align='center', height=0.5)
+            ax.set_yticks(yloc)
+            ax.set_yticklabels(features)
+            ax.set_xlabel('Importance')
+            ax.set_title('Feature Importance ({})'.format(importance_type))
+            fig.tight_layout()
+
+            tmpdir = tempfile.mkdtemp()
+            try:
+                filepath = os.path.join(tmpdir, 'feature_importance_{}.png'.format(imp_type))
+                fig.savefig(filepath)
+                try_mlflow_log(mlflow.log_artifact, filepath)
+            finally:
+                shutil.rmtree(tmpdir)
+
```

Comment:
Maybe this functions should be moved to [`mlflow/utils`](https://github.com/mlflow/mlflow/tree/master/mlflow/utils) so that we can reuse it on `xgboost.autolog`.